should I take over node-gAuth

regular opened the pull request:
Hi Ben,

I saw your node-gAuth module and took it as a starting point for a coffee-script based implementation called node-google-oauth2. I also tried to automate that first unit test that pops up a web browser. I got pretty far with automating it using phantomJS, but now I am sort of stuck. Are you interested in collaborating on this? Should I go ahead and publish on npm?

Thanks for sharing!

(oh btw, you did not specify a license, I assumed MIT, is that okay?)

Cheers!
Jan

regular committed:
fixed tests, testing with files meta data read access scope

regular committed:
listing documents works

regular committed:
changed package name to gauth, output item total of document list

regular committed:
translated to coffee, made API more verbose, added phantomJS script to automate web-based authorization (WIP)

regular committed:
changed package name

regular committed:
disabled authorizeApplication() test, enabled all other tests

regular committed:
some clean up

regular committed:
readme

regular committed:
readme

regular committed:
readme

regular committed:
cleaned up repository

regular committed:
package.json

regular committed:
readme

lyaunzbe commented:
Hey! Thanks for your contributions, I skimmed over it and it looks good overall. This was originally made to be used with a command line goog drive client I was working on, but I would love to work with you on a general google oauth2 api module, since I dont think one really exists yet.

One thing: I don't think the main body (index.js) of the module should be in coffee script. I think module code, especially that is published on npm and looked at by others, should be vanilla js. That said, I am fine with writing the tests, phantom web client, etc in coffeescript. I am a bit busy tonight, but  I will look over this code ASAP and merge when I can. I will also take a look at the automated phantom tests.

I will update the license to MIT. Thanks !

regular commented:
Yes, I did some amount on searching and could not find any module that would suit my needs. I think in the future the module should also have a command line utility that basically does what the first test does: authorizing an app for a user with given username and password.
I think if I put a little more caution into how the coffee script is written, the resulting index.js will look like hand-written. (or in my case: in fact better than hand written) :)

Ajnasz committed:
Call server close only once

Some browser will try to download favicon.ico as a second request.
After the first request the server will be closed, so the request to the
favicon.ico would generate errors in the terminal.
The patch will set a variable which value will chang after server close
called, so next handler will not be called.

regular committed:
Merge pull request #1 from Ajnasz/fix-server-close

Call server close only once

Commit:
Update google-login-phantomjs-script.coffee

regular committed:
Merge pull request #2 from nibogd/patch-1

fixed typo
